### DISCUSSION TITLE
Raspberry pi 3 + Custom datasheet + yolov5

### DISCUSSION THREAD

**Discussion by AugustodeLara** [ultralytics/yolov5#9515] [Q&A] [answered]
Hello, I’m trying to use YOLOV5 on a Raspberry pi 3.
I’m able to train my network with the default dataheat that comes in the repository.
However, when I try to train with my dataheat, which is bigger, the raspberry just doesn’t hold up and crashes during the creation of the epoch.
I’m looking for alternatives, is it possible to train my network (with my dataheat) on another machine and just run the detection on raspberry?

![image](https://user-images.githubusercontent.com/32024026/191301868-5948c4f2-b0d8-446e-99f4-2623db68c535.png)

**Comment by glenn-jocher** [comment, marked as answer]
@AugustodeLara hi buddy. RPi's are only really meant as deployment targets, they're not suitable for training. For training you can try Colab or other free GPU resources like Kaggle or Paperspace, or your own hardware.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/GCP-Quickstart)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/AWS-Quickstart)
- **Docker Image**. See [Docker Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

**Reply by AugustodeLara** [reply]
Hello @glenn-jocher !
Could I thin(python train.py) the network inside Colab and then export to raspberry ?
To just run detect ?
My changes would be since my Raspberry doesn't have much processing power.

**Reply by glenn-jocher** [reply]
@AugustodeLara yes that's right, you train in a GPU environment and then export your model to ONNX or OpenVINO for CPU inference on RPi.

**Reply by AugustodeLara** [reply]
Thanks, ! @glenn-jocher Do you have any reference documents for RPi ?

**Reply by glenn-jocher** [reply]
@AugustodeLara sorry, not yet, but we are working on creating RPi and Jetson nano tutorials in the future!

@AyushExel

**Reply by AugustodeLara** [reply]
Hello guys.
I managed to run yolov5 on Raspberry pi 3.
I trained my weights on another computer and exported an ONNX model.
It worked perfectly for detecting images and videos.
![image](https://user-images.githubusercontent.com/32024026/191645863-addfcc64-33d7-426b-98a5-d16daf0138db.png)


However, when I try to use my webcan I get an error.
I honestly don't know what it could be.

![image](https://user-images.githubusercontent.com/32024026/191646147-c3346c66-01bd-4f28-80ce-e3a7916007b4.png)


```
processor	: 0
BogoMIPS	: 38.40
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 1
BogoMIPS	: 38.40
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 2
BogoMIPS	: 38.40
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 3
BogoMIPS	: 38.40
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

Hardware	: BCM2835
Revision	: a22082
Serial		: 00000000e4f7478f
Model		: Raspberry Pi 3 Model B Rev 1.2
```
![image](https://user-images.githubusercontent.com/32024026/191646360-95d7be36-d5fc-42e8-814d-b1c0a8a6e1aa.png)

**Reply by AugustodeLara** [reply]
Update!

I managed to make it work via webcan.

I ran this command - `sudo apt-get install libxcb-util-dev`
and Afeter
I passed my `best.onnx`_(exported from another machine - because raspberry couldn't handle local training.)_  weight 
and a few more parameters and it was !

`python3 detect.py --weights best.onnx --img 640 --conf 0.1 --source 0`

**Reply by jelena2712** [reply]
@AugustodeLara how did You make Yolov5 on the videos on Raspberry Pi 3?
I would really appreciate Your answer!

**Reply by Waqas649** [reply]
Hi @AugustodeLara ,
can you please help me with code for running yolov5 model on Raspberry Pi, ?
Thanks
